File: 0078-subsets/0078-subsets.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var solve= function(nums,ans,curr,i){
    if(i==nums.length){
        ans.push([...curr]) 
        return;
    }
    curr.push(nums[i])
    solve(nums,ans,curr,i+1);
    curr.pop();
    solve(nums,ans,curr,i+1);
}
var subsets = function(nums) {
    let ans=[];
    let curr=[];
    solve(nums,ans,curr,0);
    return ans;
};

File: 0078-subsets/0078-subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>>ans;
        int i=0;
        vector<int>temp;
        solve(i,temp,ans,nums);
        return ans;

    }
    void solve(int i,vector<int>temp,vector<vector<int>>&ans,vector<int>& nums){
        if(i==nums.size()) {
            ans.push_back(temp);
            return; 
            }
            //exclude
            solve(i+1,temp,ans,nums);
            //include
            temp.push_back(nums[i]);
            solve(i+1,temp,ans,nums);


    }
};